File: redsheet-portal/src/app/database-models/project-power.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectModule, PowerTypeDetail, KnowledgeGatheringPlan,  vProjectPowerRecord  } from './'
export class ProjectPower {
    constructor(projectPower?: ProjectPower  | vProjectPowerRecord )  {
        let properties = [ "powerDetail", "projectPowerId", "projectModuleId", "powerTypeDetails", "knowledgeGatheringPlans",];
        for (let property of properties)
            if (projectPower && projectPower[property])
                this[property] = projectPower[property];
    }
 
    @required()
    @maxLength(500)
	powerDetail : string =   undefined;
 
	projectPowerId : number =   0 ;
 
    @range(0,2147483647)
	projectModuleId : number =   undefined;
	projectModule : ProjectModule  ;
	@nested(PowerTypeDetail)
	powerTypeDetails: PowerTypeDetail[];

	@nested(KnowledgeGatheringPlan)
	knowledgeGatheringPlans: KnowledgeGatheringPlan[];



}

File: RedSheet.DbEntities/Models/ProjectPower.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("ProjectPowers", Schema = "dbo")]
	[Serializable]
	public partial class ProjectPower  
	{
 
	[Required] 
	[MaxLength(500)]
	public string PowerDetail { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ProjectPowerId { get; set; } 
 
	[RelationshipTableAttribue("ProjectModules","dbo")]
	//Relationships
	public int ProjectModuleId { get; set; } 
 
	[ForeignKey("ProjectModuleId")]
	public ProjectModule ProjectModule { get; set; } 
	[InverseProperty("ProjectPower")]
	public ICollection<PowerTypeDetail> PowerTypeDetails { get; set; } 

	[InverseProperty("ProjectPower")]
	public ICollection<KnowledgeGatheringPlan> KnowledgeGatheringPlans { get; set; } 


	public ProjectPower()
	{
	this.PowerTypeDetails = new HashSet<PowerTypeDetail>();

	this.KnowledgeGatheringPlans = new HashSet<KnowledgeGatheringPlan>();

    }

    }
}

